src/misceval.js

```diff
@@ -360,63 +340,65 @@ Sk.misceval.richCompareBool = function (v, w, op, canSuspend) {
         }
     }
 
-
     // handle identity and membership comparisons
     if (op === "Is") {
-        if (v instanceof Sk.builtin.int_ && w instanceof Sk.builtin.int_) {
-            return v.numberCompare(w) === 0;
-        } else if (v instanceof Sk.builtin.float_ && w instanceof Sk.builtin.float_) {
-            return v.numberCompare(w) === 0;
-        } else if (v instanceof Sk.builtin.lng && w instanceof Sk.builtin.lng) {
-            return v.longCompare(w) === 0;
+        if (v_type === w_type) {
+            if (v === w) {
+                return true;
+            } else if (v_type === Sk.builtin.float_) {
+                return v.v - w.v === 0;
```

Comment:
this is weird 😁 why not `v.v === w.v`

Reply:
Oooh. Not sure. I think that's because v.numberCompare(w) did. Will change it